pkg/externalip/api/v1/types.go

```diff
@@ -0,0 +1,28 @@
+package v1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kapi "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// ExternalIP maps an external ip address to a service.  The object
+// name is intended to be the ip address so that name uniqueness
+// ensures an ip address is allocated at most once.
+type ExternalIP struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	// ServiceRef references the service (in any namespace) that is
+	// bound to this external ip address.
+	ServiceRef kapi.ObjectReference `json:"serviceRef"`
```

Comment:
Should be in a status struct, just like PV's.  The pattern here is exactly like PVC and PV, except we want to not make the mistake we did there (making the name a user would see confusing).  How does a user request one of these IPs?